src/bootstrap/builder.rs

```diff
@@ -150,11 +150,15 @@ impl Debug for TaskPath {
 /// Collection of paths used to match a task rule.
 #[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq)]
 pub enum PathSet {
-    /// A collection of individual paths.
+    /// A collection of individual paths or aliases.
     ///
     /// These are generally matched as a path suffix. For example, a
-    /// command-line value of `libstd` will match if `src/libstd` is in the
+    /// command-line value of `std` will match if `library/std` is in the
     /// set.
+    ///
+    /// NOTE: the paths within a set should always be aliases of one another.
+    /// For example, `src/librustdoc` and `src/tools/rustdoc` should be in the same set,
+    /// but `library/core` and `library/std` should not, even if they're executed by the same Step.
```

Comment:
```suggestion
    /// but `library/core` and `library/std` generally should not, unless there's no way (for that Step)
    /// to build them separately.
```

IOW, if a Step *does* actually alias core and std, then it seems desirable for them to be in one Set.

Reply:
Yup, that's right :) that's what `all_krates` does. (I eventually want to avoid all_krates, basically https://github.com/rust-lang/rust/pull/95503 but for all steps, but it's still supported for now.)